src/components/tl/mlx5/mcast/tl_mlx5_mcast_helper.h

```diff
@@ -0,0 +1,320 @@
+/**
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5.h"
+
+ucc_status_t ucc_tl_probe_ip_over_ib(const char* ib_dev_list, struct
+                                     sockaddr_storage *addr);
+
+#ifndef MCAST_H_
+#define MCAST_H_
+#include "tl_mlx5_mcast_progress.h"
+
+#define MIN(a,b) (((a)<(b))?(a):(b))
+#define MAX(a,b) (((a)>(b))?(a):(b))
+
+static inline int ucc_tl_mlx5_mcast_poll_send(ucc_tl_mlx5_mcast_coll_comm_t *comm)
+{
+    struct ibv_wc wc;
+    int           num_comp;
+    
+    num_comp = ibv_poll_cq(comm->scq, 1, &wc);
+    
+    tl_trace(comm->lib, "polled send completions: %d", num_comp);
+    
+    if (num_comp < 0) {
+
+        tl_error(comm->lib, "send queue poll completion failed %d", num_comp);
+
+    } else if (num_comp > 0) {
+
+        if (IBV_WC_SUCCESS != wc.status) {
+           tl_warn(comm->lib, "mcast_poll_send: %s err %d num_comp",
+                    ibv_wc_status_str(wc.status), num_comp);
+            return -1;
+        }
+
+        comm->pending_send -= num_comp;
+    }
+    
+    return num_comp;
+}
+
+static inline ucc_status_t ucc_tl_mlx5_mcast_send(ucc_tl_mlx5_mcast_coll_comm_t *comm,
+                                                  ucc_tl_mlx5_mcast_coll_req_t *req,
+                                                  int num_packets, const int zcopy)
+{
+    struct ibv_send_wr *swr            = &comm->mcast.swr;
+    struct ibv_sge     *ssg            = &comm->mcast.ssg;
+    int                 max_per_packet = comm->max_per_packet;
+    int                 offset         = req->offset, i;
+    struct ibv_send_wr *bad_wr;
+    struct pp_packet   *pp;
+    int                 rc;
+    int                 length;
+
+    for (i = 0; i < num_packets; i++) {
+        pp = ucc_tl_mlx5_mcast_buf_get_free(comm);
+        ucc_assert(pp->context == 0);
+
+        __builtin_prefetch((void*) pp->buf);
+        __builtin_prefetch(req->ptr + offset);
```

Comment:
are those prefetches really needed?

Reply:
This is a legacy code snipped from the hcoll. I have not done any performance testing to see how much impact they have in total mcast latency. But yes, we can remove them safely.